src/fsharp/FSharp.Core/local.fs

```diff
@@ -557,6 +557,33 @@ module internal List =
 
             loop 0 (windowSize - 1) list
 
+    let rec chunksOfToFreshConsTail chunkCons resCons chunkSize i list =
+        match list with
+        | [] ->
+            setFreshConsTail chunkCons []
+            setFreshConsTail resCons []
+        | h::t ->
+            let cons = freshConsNoTail h
+            if i = chunkSize then
+                setFreshConsTail chunkCons []
+                let newResCons = freshConsNoTail cons
+                setFreshConsTail resCons newResCons
+                chunksOfToFreshConsTail cons newResCons chunkSize 1 t
+            else
+                setFreshConsTail chunkCons cons
+                chunksOfToFreshConsTail cons resCons chunkSize (i+1) t
+            ()
```

Comment:
Is this final `()` needed? All paths result in `unit` recursive calls as far as I can see. If it's not needed, then maybe all it's doing is preventing func from being tail recursive :rabbit2: ->  :turtle: